test/cql-pytest/test_validation.py

```diff
@@ -330,11 +330,7 @@ def test_validation_ascii_query(cql, table1):
         assert results[0].k == 1 and results[0].a == s
     for s in bad_ascii:
         print(s)
-        # Scylla prints "marshaling error: Value not compatible with type
-        # org.apache.cassandra.db.marshal.AsciiType: '...'". Cassandra prints
-        # "Invalid ASCII character in string literal". The only thing in common
-        # is the word "ascii"...
-        with pytest.raises(InvalidRequest, match=re.compile('ascii', re.IGNORECASE)):
+        with pytest.raises(InvalidRequest, match='Invalid ASCII character'):
```

Comment:
The same marshaling error thing exists in the `test_validation_ascii_bound_column` test function. Can you please change it too? Did you patch actually fix the bound value case as well?

(BTW, although these changes are not "needed", in the sense that the new message also passes the old test - but it's indeed very good to make the test more strict. You can even add a "Reproduces issue #14320" comment to these tests, which will allow us in the future to understand that this test fails because we forgot to backport the fix for that issue.)

Reply:
> The same marshaling error thing exists in the `test_validation_ascii_bound_column` test function. Can you please change it too? Did you patch actually fix the bound value case as well?

no. this patch does not fix the bound value case. one would need to update `validate_visitor` in `type/types.cc`.

the existing error message reads: "Validation failed - non-ASCII character in an ASCII string" is better than Cassandra's error of "Invalid byte for ascii: ..". i have no compelling reason to change it unless we really want to have exactly the same error as Cassandra. but i don't see it.

> 
> (BTW, although these changes are not "needed", in the sense that the new message also passes the old test - but it's indeed very good to make the test more strict.

yeah, and the comment does not apply anymore. 

> You can even add a "Reproduces issue #14320" comment to these tests, which will allow us in the future to understand that this test fails because we forgot to backport the fix for that issue.)

sure.

Reply:
> > The same marshaling error thing exists in the `test_validation_ascii_bound_column` test function. Can you please change it too? Did you patch actually fix the bound value case as well?
> 
> no. this patch does not fix the bound value case. one would need to update `validate_visitor` in `type/types.cc`.

Could you please consider doing that case too? If we don't do it now, we'll never remember to do it later :-(

Reply:
@nyh hi Nadav, i explained the reason why i liked existing error message better. could you help me understand why we need to "do that case too"? BTW, "ascii" is more correct than "ASCII", IMHO.

Reply:
You're right, I missed that explanation. I agree that our different message in that case is perfectly reasonable, so let's keep it.

On the "BTW", I am guessing you are saying that ASCII (which we print) is more correct than ascii (that cassandra prints), and I agree.